packages/@aws-cdk/aws-stepfunctions/lib/condition.ts

```diff
@@ -177,8 +177,8 @@ enum CompoundOperator {
 class VariableComparison extends Condition {
     constructor(private readonly variable: string, private readonly comparisonOperator: ComparisonOperator, private readonly value: any) {
         super();
-        if (!variable.startsWith('$.')) {
-            throw new Error(`Variable reference must start with '$.', got '${variable}'`);
+        if (!variable.startsWith('$.') && !variable.startsWith('$[')) {
```

Comment:
I was curious about the performance implications of using two `startsWith` vs. a single RegExp, and it [seems like the latter is 57% faster on Chrome 77](https://jsperf.com/str-startswith-vs-regex).

Not a change request by any means, just sharing 😄

Reply:
Thanks, even faster when using `.test` instead of `.match`: https://jsperf.com/str-startswith-vs-regex/2, updated PR

Reply:
Oh wow, I didn't expect the gain to be this big with `.test(`.
I also picked a worst case scenario for `startsWith` using `$a` as a test.

But anyway, thanks for nitpick-y change 👍